test/e2e_federation/upgrades/generic.go

```diff
@@ -58,8 +59,13 @@ func (t *GenericUpgradeTest) Teardown(f *fedframework.Framework) {
 	// Rely on the namespace deletion to clean up everything
 }
 
-// SecretUpgradeTest validates that a secret remains propagated before
-// and after a controlplane upgrade
-func SecretUpgradeTest() Test {
-	return &GenericUpgradeTest{adapter: &typeadapters.SecretAdapter{}}
+func GenericUpgradeTests() []Test {
```

Comment:
I'm a stickler for comments, for several reasons:
- they make the code look more polished, which increases my confidence in its quality
- when writing them, they force me to consider whether I really understand what the thing I'm describing is/is doing
- when writing them, they force me to consider whether the thing I'm describing needs to be two things (e.g., if my comment is "do this and that", maybe I have two functions smushed into one)
- they give a little more context to someone reading the code for the first time
- they help preserve context that seemed obvious but actually wasn't
- they help suggest better function/type/variable/constant names

Of course, not every comment actually accomplishes all of these ends, and there are risks to putting comments on things that don't really need them, but I'm inclined to believe that I'll forget what "GenericUpgradeTests()" means at some point and will want some description.